apps/spruce/src/components/VersionRestartModal/VersionRestartModal.tsx

```diff
@@ -167,6 +177,19 @@ const VersionRestartModal: React.FC<VersionRestartModalProps> = ({
               <br />
             </div>
           )}
+
+          {selectedTotal + numEstimatedActivatedGeneratedTasks >
+            largeNumFinalizedTasksThreshold && (
+            <Banner data-cy="disabled-webhook-banner" variant="warning">
+              This is a large operation, expected to schedule{" "}
+              {selectedTotal + numEstimatedActivatedGeneratedTasks} tasks.
+              Please confirm that this number of tasks is necessary before
+              continuing. For more information, please refer to our{" "}
+              <StyledLink href={taskSchedulingLimitsDocumentationUrl}>
+                docs.
+              </StyledLink>
+            </Banner>
+          )}
```

Comment:
If it's not too much trouble I think moving this banner into its own component (like `components/TaskSchedulingWarning`) would be ideal since it's exactly the same across the three components!